Abstract Syntax and Semantics Document/index.html

```diff
@@ -500,53 +500,159 @@
+        a finite set <code>w<sub>P</sub></code> of timestamp predicates together with
+        semi-infinite temporal intervals <code>w<sub>J</sub>(P)</code> of each timestamp predicate <code>P</code> in <code>w<sub>P</sub></code>
+        with endpoints <code>w<sub>T</sub>(P)</code>,
+        and positive integers <code>w<sub>N</sub>(P)</code>, for each timestamp predicate <code>P</code> in <code>w<sub>P</sub></code> such that
+        for every stream <code>S</code>, 
+        an element <code>S(i)</code> of <code>S</code> is a member of <code>w(S)</code> if and only if
+        the timestamp predicate  <code>P</code> of <code>S(i)</code> is in <code>P</code> and
+        <ul>          
+          <li><code>t</code> is the timestamp of <code>S(i)</code>,</li>
+          <li><code>T</code> is the set of temporal entities that are timestamps for elements of <code>S</code>
+            with timestamp <code>P</code> that belong to <code>w<sub>J</sub>(P)</code>
+            and belong to the closed interval between <code>t</code> and <code>w<sub>T</sub>(P)</code>, and</li>
+          <li>the cardinality of <code>T</code> is less than or equal to  <code>w<sub>N</sub>(P)</code></li>
+          </li>
+        </ul>
+      </p>
+      
+      <aside class="note">The above definition of temporal-count-based window function allows the following characteristics
+        to be set indepndently for each timestamp predicate of concern:
+        <ul>
+          <li>future-facing or past-facing orientation,</li>
+          <li>inclusive or exclusive temporal anchor,</li>
+          <li>depth.</li>
+        </ul>
+      </aside>
+      
+      <p class="note">The substream resulting from the application of a temporal-count-based window function is temporal-count-bounded.</p>
+      
+      <p class="note">Due to the potential for stream elements with incomparable or duplicate timestamps, the number of elements in the substream having a particular
+        timestamp predicate
+      is not guaranteed to be equal to the depth specified for the predicate by the temporal-count-based window. </p>
+      
+      <p class="ednote">
+       Applications that require a substream with an exact number of elements for a specified timestamp predicate
+       might randomly discard elements after applying a temporal-count-based window function, according to some criterion, e.g.  
+       extreme elements (maximal or minimal, depending on the orientation of the counting).
+      However, this extra step causes the process to be nondeterministic, and hence is not a function 
+      (a problem that might be resolved by defining a window relation instead of a function.)
+      If elements are discarded in a nonrandom fashion based on their order in the stream sequence, then this
+      would be a function, but would not preserve isomorphism, and so would not be a window function. 
+      </p>
```

Comment:
What's in there now is called "length-based window relation" or
"length-based window function". I am happy to change that to
"element-count-based" as it seems more symmetrical to
"temporal-count-based". There is a separate issue, on the other document
actually
https://github.com/streamreasoning/RSP-QL/issues/64#issuecomment-222165487,
about "element" being the terminology for a single "thingie" of the stream
sequence. I suggested "item", "element", or "member". Whatever gets
selected, then we should use that for this window function, I believe.

Tara

On Fri, May 27, 2016 at 3:37 PM, beta2k <notifications@github.com> wrote:

> In Abstract Syntax and Semantics Document/index.html
> <https://github.com/streamreasoning/RSP-QL/pull/42#discussion_r64905218>:
>
> > +
> > +      <p class="note">The substream resulting from the application of a temporal-count-based window function is temporal-count-bounded.</p>
> > +
> > +      <p class="note">Due to the potential for stream elements with incomparable or duplicate timestamps, the number of elements in the substream having a particular
> > +        timestamp predicate
> > +      is not guaranteed to be equal to the depth specified for the predicate by the temporal-count-based window. </p>
> > +
> > +      <p class="ednote">
> > +       Applications that require a substream with an exact number of elements for a specified timestamp predicate
> > +       might randomly discard elements after applying a temporal-count-based window function, according to some criterion, e.g.
> > +       extreme elements (maximal or minimal, depending on the orientation of the counting).
> > +      However, this extra step causes the process to be nondeterministic, and hence is not a function
> > +      (a problem that might be resolved by defining a window relation instead of a function.)
> > +      If elements are discarded in a nonrandom fashion based on their order in the stream sequence, then this
> > +      would be a function, but would not preserve isomorphism, and so would not be a window function.
> > +      </p>
>
> I like element-count-based windows. Would this require a precise
> definition of what an element actually is somehwere?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/streamreasoning/RSP-QL/pull/42/files/06507b6e1f5cec50c3fde474a72885912c68201a#r64905218>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABMMVd5PoGEBYbi5gGBldgCaqK0p2t7aks5qFvOjgaJpZM4HX2bR>
> .
>